pkg/operator/ceph/object/spec.go

```diff
@@ -489,13 +499,50 @@ func (c *clusterConfig) defaultReadinessProbe() (*v1.Probe, error) {
 	return probe, nil
 }
 
+func getRGWProbePathAndCode(protocolSpec cephv1.ProtocolSpec) (string, int) {
+	enabledAPIs := buildRGWEnableAPIsConfigVal(protocolSpec)
+	if len(enabledAPIs) == 0 {
+		// all apis including s3 are enabled
+		// using default s3 Probe
+		return "", 0
+	}
+	if slices.Contains(enabledAPIs, "s3") {
+		// using default s3 Probe
+		return "", 0
+	}
+	if slices.Contains(enabledAPIs, "swift") {
+		// using swift api for probe
+		// calculate path for swift probe
+		prefix := "/swift/"
+		if protocolSpec.Swift != nil && protocolSpec.Swift.UrlPrefix != nil && *protocolSpec.Swift.UrlPrefix != "" {
+			prefix = *protocolSpec.Swift.UrlPrefix
+			if !strings.HasPrefix(prefix, "/") {
+				prefix = "/" + prefix
+			}
+			if !strings.HasSuffix(prefix, "/") {
+				prefix += "/"
+			}
+		}
+		prefix += "info"
+		return prefix, 0
+	}
+	if slices.Contains(enabledAPIs, "admin") {
+		// using admin-ops api for probe
+		// check that admin ops info responds with 403 - Forbidden because all admin endpoinds secured
+		return "/admin/info", 403
+	}
+	logger.Warningf("no suitable for probe api enabled %v. Fallback to s3 probe", enabledAPIs)
```

Comment:
Is this better wording?

```suggestion
	logger.Warningf("no suitable api for probe enabled %v. Fallback to s3 probe", enabledAPIs)
```